commit 2b81ec3265bec26f728b1733faff9151dcd496b9
Author: Jay Foad <jay.foad@amd.com>
Date:   Fri Apr 21 13:59:13 2023 +0100

    Revert "[ConstantFolding] Fix crash when folding vector llvm.is.fpclass"
    
    This reverts commit 5fc6425fb6c77052a26cf0cf7b886449fabe1af4.
    
    It is reported to cause other crashes that require a larger fix.

diff --git a/llvm/lib/Analysis/VectorUtils.cpp b/llvm/lib/Analysis/VectorUtils.cpp
index d6a760b8571e..7700c722765d 100644
--- a/llvm/lib/Analysis/VectorUtils.cpp
+++ b/llvm/lib/Analysis/VectorUtils.cpp
@@ -86,7 +86,6 @@ bool llvm::isTriviallyVectorizable(Intrinsic::ID ID) {
   case Intrinsic::pow:
   case Intrinsic::fma:
   case Intrinsic::fmuladd:
-  case Intrinsic::is_fpclass:
   case Intrinsic::powi:
   case Intrinsic::canonicalize:
   case Intrinsic::fptosi_sat:
@@ -104,7 +103,6 @@ bool llvm::isVectorIntrinsicWithScalarOpAtArg(Intrinsic::ID ID,
   case Intrinsic::abs:
   case Intrinsic::ctlz:
   case Intrinsic::cttz:
-  case Intrinsic::is_fpclass:
   case Intrinsic::powi:
     return (ScalarOpdIdx == 1);
   case Intrinsic::smul_fix:
diff --git a/llvm/test/Transforms/InstSimplify/is_fpclass.ll b/llvm/test/Transforms/InstSimplify/is_fpclass.ll
deleted file mode 100644
index b14bfcbbfaac..000000000000
--- a/llvm/test/Transforms/InstSimplify/is_fpclass.ll
+++ /dev/null
@@ -1,12 +0,0 @@
-; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 2
-; RUN: opt < %s -S -passes=instsimplify | FileCheck %s
-
-define <2 x i1> @f() {
-; CHECK-LABEL: define <2 x i1> @f() {
-; CHECK-NEXT:    ret <2 x i1> zeroinitializer
-;
-  %i = call <2 x i1> @llvm.is.fpclass.v2f16(<2 x half> <half 0xH7C00, half 0xH7C00>, i32 3)
-  ret <2 x i1> %i
-}
-
-declare <2 x i1> @llvm.is.fpclass.v2f16(<2 x half>, i32 immarg)
